tests/integ/modin/test_apply_snowpark_python_functions.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import math
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+
+from tests.integ.modin.utils import assert_frame_equal, assert_series_equal
+from tests.integ.utils.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=4)
+def test_apply_sin():
+    from snowflake.snowpark.functions import sin
+
+    native_s = native_pd.Series([0.00, -1.23, 10, math.pi, math.pi / 2])
+    s = pd.Series(native_s)
+
+    assert_series_equal(s.apply(sin), native_s.apply(math.sin))
+    assert_series_equal(s.map(sin), native_s.map(math.sin))
+    assert_frame_equal(
+        s.to_frame().applymap(sin), native_s.to_frame().applymap(math.sin)
+    )
+    assert_frame_equal(
+        s.to_frame().apply(sin),
+        native_s.to_frame().apply(np.sin),  # Note math.sin does not work with df.apply
+    )
+
+
+@sql_count_checker(query_count=0)
+def test_apply_snowpark_python_function_not_implemented():
```

Comment:
should this test be parametrized instead?

Reply:
not necessary because it is super fast. No I/O or query execution.